File: 04_조준장/10828.py
import sys


n = int(sys.stdin.readline())
stack = []

# 반복문을 통해 명령을 수행
for i in range(n):
    # 리스트 형식으로 명령을 받는다.
    order = list(map(str, sys.stdin.readline().split()))

    if order[0] == "pop":
        if stack:
            print(stack.pop())
        else:
            print(-1)
            
    elif order[0] == "size":
        print(len(stack))

    elif order[0] == "empty":
        if stack:
            print(0)
        else:
            print(1)
            
    elif order[0] == "top":
        if stack:
            print(stack[-1])
        else:
            print(-1)
    
    else:
        stack.append(order[1])

File: 08_김채린/10828.c
#include <stdio.h>
#include <string.h>
int stack[10000];
int stack_size = 0;

void push(int x) {
    stack[stack_size] = x;
    stack_size += 1;
}

int empty() {
    if (stack_size == 0) {
        return 1;
    }
    return 0;
}

int pop() {
    if (empty()) {
        return -1;
    }
    stack_size -= 1;
    return stack[stack_size];
}

int top() {
    if (empty()) {
        return -1;
    }
    return stack[stack_size - 1];
}

int main() {
    int N = 0, x = 0;
    char str[10] = { 0, };
    scanf("%d", &N);

    for (int i = 0; i < N; i++) {
        scanf("%s", str);
        if (!strcmp(str, "push")) {
            scanf("%d", &x);
            push(x);
        }
        else if (!strcmp(str, "pop")) {
            printf("%d\n", pop());
        }
        else if (!strcmp(str, "empty")) {
            printf("%d\n", empty());
        }
        else if (!strcmp(str, "size")) {
            printf("%d\n", stack_size);
        }
        else if (!strcmp(str, "top")) {
            printf("%d\n", top());
        }
    }
    return 0;
}